xdsl/transforms/convert_stencil_to_csl_stencil.py

```diff
@@ -55,69 +67,37 @@ def get_stencil_access_operands(op: Operand) -> set[Operand]:
     return res
 
 
-@dataclass(frozen=True)
-class RestructureSymmetricReductionPattern(RewritePattern):
-    """
-    Consume data first where that data comes from stencil accesses to `buf`.
+def _get_prefetch_buf(op: stencil.ApplyOp) -> int | None:
+    # calculate memory cost of all prefetch operands
+    def get_prefetch_overhead(o: OpResult):
+        assert isa(o.type, TensorType[Attribute])
+        buf_count = o.type.get_shape()[0]
+        buf_size = prod(o.type.get_shape()[1:])
+        return buf_count * buf_size
```

Comment:
```suggestion
        return prod(o.type.get_shape())
```

This appears to just do a product of all dimensions?

Reply:
True, slightly favour the readability here. Changed anyway.